protocol/x/vault/keeper/orders_test.go

```diff
@@ -1 +1,424 @@
+							clobtypes.Order_SIDE_BUY,
+							uint8(2),
+						),
+						OrderFlags: clobtypes.OrderIdFlags_LongTerm,
+						ClobPairId: constants.Vault_Clob_1.Number,
+					},
+					Side:     clobtypes.Order_SIDE_BUY,
+					Quantums: testClobPairs[1].StepBaseQuantums, // TODO (TRA-144): Implement order size
+					Subticks: lib.BigRatRoundToNearestMultiple(
+						// subticks of bid_2 = subticks * (1 - spread)^2
+						lib.BigRatMulPpm(
+							lib.BigRatMulPpm(ethSubticks, lib.OneMillion-ethSpreadPpm),
+							lib.OneMillion-ethSpreadPpm,
+						),
+						testClobPairs[1].SubticksPerTick,
+						false, // round down for bids
+					),
+					GoodTilOneof: orderGtbt,
+				},
+			},
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			vaultOrders := tApp.App.VaultKeeper.GetVaultClobOrders(
+				ctx,
+				tc.vaultId,
+			)
+			require.Equal(t, tc.expectedOrders, vaultOrders)
+		})
+	}
+}
+
+func TestGetVaultClobOrderClientId(t *testing.T) {
+	tests := map[string]struct {
+		/* --- Setup --- */
+		// side.
+		side clobtypes.Order_Side
+		// block height.
+		blockHeight int64
+		// layer.
+		layer uint8
+
+		/* --- Expectations --- */
+		// Expected client ID.
+		expectedClientId uint32
+	}{
+		"Buy, Block Height Odd, Layer 1": {
+			side:             clobtypes.Order_SIDE_BUY, // 0<<31
+			blockHeight:      1,                        // 1<<30
+			layer:            1,                        // 1<<22
+			expectedClientId: 0<<31 | 1<<30 | 1<<22,
+		},
+		"Buy, Block Height Even, Layer 1": {
+			side:             clobtypes.Order_SIDE_BUY, // 0<<31
+			blockHeight:      2,                        // 0<<30
+			layer:            1,                        // 1<<22
+			expectedClientId: 0<<31 | 0<<30 | 1<<22,
+		},
+		"Sell, Block Height Odd, Layer 2": {
+			side:             clobtypes.Order_SIDE_SELL, // 1<<31
+			blockHeight:      1,                         // 1<<30
+			layer:            2,                         // 2<<22
+			expectedClientId: 1<<31 | 1<<30 | 2<<22,
+		},
+		"Sell, Block Height Even, Layer 2": {
+			side:             clobtypes.Order_SIDE_SELL, // 1<<31
+			blockHeight:      2,                         // 0<<30
+			layer:            2,                         // 2<<22
+			expectedClientId: 1<<31 | 0<<30 | 2<<22,
+		},
+		"Buy, Block Height Even, Layer Max Uint8": {
+			side:             clobtypes.Order_SIDE_BUY, // 0<<31
+			blockHeight:      123456,                   // 0<<30
+			layer:            math.MaxUint8,            // 255<<22
+			expectedClientId: 0<<31 | 0<<30 | 255<<22,
+		},
+		"Sell, Block Height Odd, Layer 0": {
+			side:             clobtypes.Order_SIDE_SELL, // 1<<31
+			blockHeight:      12345654321,               // 1<<30
+			layer:            0,                         // 0<<22
+			expectedClientId: 1<<31 | 1<<30 | 0<<22,
+		},
+		"Sell, Block Height Odd (negative), Layer 202": {
+			side:             clobtypes.Order_SIDE_SELL, // 1<<31
+			blockHeight:      -678987,                   // 1<<30
```

Comment:
nit: Weird test case given this is definitely an invalid state for the ctx.BlockHeight to be negative. Non-blocking though, maybe note that somewhere?

Reply:
yeah just added this test case as `blockHeight` is an `int64`. wanted to make sure that a negative value doesn't break this function. will add a note on this